### PR TITLE
Dedup js source list in BUILD.gn

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -52,6 +52,31 @@ main_extern = [
...
+  "js/remove.ts",
+  "js/stat.ts",
+  "js/text_encoding.ts",
+  "js/timers.ts",
+  "js/tsconfig.generated.json",
+  "js/types.ts",
+  "js/util.ts",
+  "js/v8_source_maps.ts",
+  "js/write_file.ts",
+  "tsconfig.json",
```
Comment (qti3e) [inline BUILD.gn]: Add `js/main.ts`

File: BUILD.gn [hunk 2]
```diff
@@ -186,29 +211,7 @@ executable("snapshot_creator") {
...
-    "js/stat.ts",
-    "js/text_encoding.ts",
-    "js/timers.ts",
-    "js/tsconfig.generated.json",
-    "js/types.ts",
-    "js/util.ts",
-    "js/v8_source_maps.ts",
-    "js/write_file.ts",
-  ]
+  sources = ts_sources
```
Comment (qti3e) [inline BUILD.gn]: Probably you want to add `+ ["js/plugins.d.ts" + "js/fetch_types.d.ts"]` here.

File: BUILD.gn (hunk 1 above)
Comment (ry) [reply BUILD.gn]: Thanks fixed.

File: BUILD.gn (hunk 2 above)
Comment (ry) [reply BUILD.gn]: Fixed.

**Comment by piscisaureus** [comment]
I see that you also added some previously missing files (libdeno.ts and mock_builtin.ts).
This is a recurring source of frustration. It would be very valuable if we could could catch it when bundle inputs are not properly listed in BUILD.gn. 
LGTM

**Pull request merged**